neevaco/neeva-android
# Work around about:blank tabs being created

When the user clicks on a link and it opens a new tab, it's possible for WebLayer to end up firing an Intent to open the link in another app.  When this happens, WebLayer automatically closes the tab but it sets the active tab back to null.  This is problematic because WebLayer sends the user to about:blank when they come back to our app.

To work around this, detect when the user is coming back to our app without an active tab set and set it to whatever the most recently used tab was -- unless the user was in the tab switcher.  If the user was in the tab switcher, we preserve the existing behavior of letting no active tab be set.  If the app dies with no active tab set, WebLayer will automatically pick the most recently created tab.

## Checklists

### How was this tested?

### Manual test cases
1. Install the Reddit app
2. In our app, do a Neeva search for "dog"
3. Click on a link that would open Reddit (under one of the mentioned by links)
4. See that the app briefly creates an about:blank tab before sending the user to Reddit
5. Go back to our app
6. See that the user is back on the search page that spawned the about:blank page

### Screenshots and screen recordings

https://user-images.githubusercontent.com/96092902/199365308-2200ed92-64cc-4c60-bd8a-a31ec02fe3e2.mp4

https://user-images.githubusercontent.com/96092902/199365318-b1951fc2-ca39-4029-8ac8-2b783784c5ef.mp4



## Issues addressed
* Related to https://github.com/neevaco/neeva-android/issues/807
